### PR TITLE
lonn plugins

### PR DESCRIPTION
**Pull request by Mnstrman06**
TODO: arrow bubble texture support

### CODE REVIEW

File: Loenn/lang/en_gb.lang [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Mod Name
+mods.EverestYamlName.name=IsaGrabBag
```
Comment (catapillie) [inline Loenn/lang/en_gb.lang]: You probably forgot to change `EverestYamlName`, which is supposed to match the mod's `everest.yaml` name. Whatever you write after the equal sign is what is shown in Lönn.
```suggestion
mods.IsaGrabBag.name=Isa's Grab Bag
```

**Review by catapillie — CHANGES_REQUESTED** [review]
Please use `en_gb.lang` to give name to placements.

For example, let's say you are working on the `isaBag/arrowBubble` plugin. You have a placement for each direction, up, down, left and right. Instead of hardcoding the "human-readable" placements names into the plugin's code, give them a simple name like `"left"` for the left placements, `"down"` for the down one, etc... you get the idea.

You'll then want to associate these placement names to an actual readable name using the language file. So, in our example, it would be something like:
```txt
entities.isaBag/arrowBubble.placements.name.up=Arrow Bubble (Up)
entities.isaBag/arrowBubble.placements.name.down=Arrow Bubble (Down)
entities.isaBag/arrowBubble.placements.name.left=Arrow Bubble (Left)
entities.isaBag/arrowBubble.placements.name.right=Arrow Bubble (Right)
```

You don't need to mention `IsaGrabBag` in the placements names. Lönn will add it itself, given you defined the name in the language file. For example, in this case, it'd be:
```txt
mods.IsaGrabBag.name=Isa's Grab Bag
```
(Or, you know, you get to format it the way you want. I also like to put this at the top of the language file.)

File: Loenn/entities/arrowBubble.lua [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+        }
+    },
+        name = "Arrow Bubble (Down, IsaGrabBag)",
+        data = {
+            direction = down
+        }
+    }
+}
+
+#TODO: add texture based on direction
```
Comment (bigkahuna443) [inline Loenn/entities/arrowBubble.lua]: all the images are formatted as `booster{dir}00.png`, so you can use something like this:
`local texture = "isafriend/objects/booster/booster" .. dir .. "00"`

File: Loenn/entities/arrowBubble.lua [hunk 3]
```diff
@@ -0,0 +1,31 @@
+local arrowBubble = {}
+
+arrowBubble.name = "isaBag/arrowBubble"
+arrowBubble.depth = -100
+arrowBubble.placements = {
+    {
+        name = "Arrow Bubble (Left, IsaGrabBag)",
```
Comment (bigkahuna443) [inline Loenn/entities/arrowBubble.lua]: to follow up on catapillie's comment, you would change this line to be `name = "left"` and then have `[...]placements.name.left = "Arrow Bubble (Left)"` in the en_gb.lang. check out some of the other legacy loenn plugin repos (or vanilla) for examples

File: Loenn/entities/arrowBubble.lua [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+        data = {
+            direction = left
```
Comment (bigkahuna443) [inline Loenn/entities/arrowBubble.lua]: This doesn't load lol, looks like it's WIP though. I'd recommend making the `direction` field a non-editable dropdown with the directions. Looks like the code needs it to be lowercase which is annoying, I think there's a way to make it display separately though

File: Loenn/entities/rewind.lua [hunk 5]
```diff
@@ -0,0 +1,11 @@
+local rewindCrystal = {}
+
+rewindCrystal.name = "isaBag/rewindCrystal"
+rewindCrystal.depth = -100
+rewindCrystal.texture = "isafriend/objects/rewind/idle00"
+rewindCrystal.placements = {
+    name = "Rewind Crystal (IsaGrabBag)",
+    oneUse = true
```
Comment (bigkahuna443) [inline Loenn/entities/rewind.lua]: should be in a `data` table

File: Loenn/entities/waterBoost.lua [hunk 6]
```diff
@@ -0,0 +1,11 @@
+local waterBoost = {}
+
+waterBoost.name = "isaBag/waterBoost"
+waterBoost.depth = 0
+waterBoost.texture = "isafriend/helperimage"
+waterBoost.placements = {
+    name = "Water Boost Controller (IsaGrabBag)",
+    boostEnabled = true
```
Comment (bigkahuna443) [inline Loenn/entities/waterBoost.lua]: should be in a `data` table

File: Loenn/entities/zipline.lua [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+zipline.name = "isaBag/zipline"
+zipline.depth = 0
+zipline.texture = "isafriend/objects/zipline/handle"
+zipline.nodeTexture = "isafriend/objects/zipline/handle_end"
+zipline.nodeLimits = {0, -1}
+zipline.nodeVisibility = "always"
+zipline.nodeLineRenderType = "fan"
+zipline.placements = {
+    name = "Zipline (IsaGrabBag)",
+    useStamina = true
```
Comment (bigkahuna443) [inline Loenn/entities/zipline.lua]: should be in a `data` table

File: Loenn/triggers/coreWindTrigger.lua [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    data = {
+        patternCold = "Down",
+        patternHot = "Up"
+    }
+}
+
+coreWindTrigger.fieldInformation = {
+    patternCold = {
+        options = {
+            "None",
```
Comment (bigkahuna443) [inline Loenn/triggers/coreWindTrigger.lua]: use the `wind_patterns` enum for these

File: Loenn/lang/en_gb.lang [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+# Arrow Bubble
+entities.isaBag/arrowBubble.placements.name.default=arrow Bubble
```
Comment (bigkahuna443) [inline Loenn/lang/en_gb.lang]: please use consistent capitalization for these lol. so "arrow Bubble" -> "Arrow Bubble"

File: Loenn/entities/zipline.lua [hunk 10]
```diff
@@ -0,0 +1,15 @@
+local zipline = {}
+
+zipline.name = "isaBag/zipline"
+zipline.depth = 0
+zipline.texture = "isafriend/objects/zipline/handle"
+zipline.nodeTexture = "isafriend/objects/zipline/handle_end"
+zipline.nodeLimits = {0, -1}
+zipline.nodeVisibility = "always"
+zipline.nodeLineRenderType = "fan"
```
Comment (bigkahuna443) [inline Loenn/entities/zipline.lua]: not sure what these choices are for nodes. use `{0, 2}` for limits, and it should be rendering the actual line like in the ahorn plugin

**Review by bigkahuna443 — CHANGES_REQUESTED** [review]
Overall notes:
1. Please test plugins in-editor before making a PR, would've caught a lot of these.
2. Missing plugins for the Arrow Block, Corner Boost Block, and the Force Variant Trigger, not sure if you intended to add them or forgot. The latter two should not get placements but still need plugins

**Comment by catapillie** [comment]
Is this still being worked on? I'm willing to review the code, and possibly improve the ports if necessary.

**Comment by Mnstrman06** [comment]
The ports are not finished, the remaining functions are things outside of my very limited skillset.

**Pull request closed without merging**